### PR TITLE
Module skeleton

### PR DESCRIPTION
**Pull request by anantjain6**

### CODE REVIEW

**Comment by ankitjain28may** [comment]
Integrate tugboat, write config for it.

**Comment by dbjpanda** [comment]
> Integrate tugboat, write config for it.

Create another repo "Drupal", make a mirror/subversion/subtree of the repo 'collaborative_editors.' inside Drupal/web/modules/. Push your entire configuration, db-dump etc to 'Drupal' repo. and set up tugboat for that repo. Use https://github.com/drupal-composer/drupal-project to set up your Drupal installation. Hope it's clear.

**Comment by dbjpanda** [comment]
Also rename those submodules from `collaborative_editor_etherpad` to only etherpad. The name seems very long. You may use a small prefix like `ce`. E.g ce_etherpad

**Comment by ankitjain28may** [comment]
Don't setup tugboat just for the modules repo, it won't work there.

**Comment by anantjain6** [comment]
> Also rename those submodules from `collaborative_editor_etherpad` to only etherpad. The name seems very long. You may use a small prefix like `ce`. E.g ce_etherpad

"ce_etherpad" looks better to me since "etherpad" machine name is already reserved in drupal.org
Added commit for the same

**Comment by ankitjain28may** [comment]
> > Integrate tugboat, write config for it.
> 
> Create another repo "Drupal", make a mirror/subversion/subtree of the repo 'collaborative_editors.' inside Drupal/web/modules/. Push your entire configuration, db-dump etc to 'Drupal' repo. and set up tugboat for that repo. Use https://github.com/drupal-composer/drupal-project to set up your Drupal installation. Hope it's clear.

Do this, so we can review.

File: modules/ce_etherpad/ce_etherpad.routing.yml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+ce_etherpad.settings:  
+  path: '/admin/config/collaborative-editor/etherpad'
```
Comment (ankitjain28may) [inline modules/ce_etherpad/ce_etherpad.routing.yml:2]: collaborative-editors

File: modules/ce_etherpad/src/Controller/EtherpadController.php [hunk 2]
```diff
@@ -0,0 +1,12 @@
+<?php
```
Comment (ankitjain28may) [inline modules/ce_etherpad/src/Controller/EtherpadController.php:1]: This file isn't required for now.

File: src/Form/CollaborativeEditorsSettingsForm.php [hunk 3]
```diff
@@ -0,0 +1,13 @@
+<?php
```
Comment (ankitjain28may) [inline src/Form/CollaborativeEditorsSettingsForm.php:1]: This file is also not required as of now.

File: collaborative_editors.links.menu.yml [hunk 4]
```diff
@@ -0,0 +1,13 @@
+collaborative_editors.admin_config:
```
Comment (ankitjain28may) [inline collaborative_editors.links.menu.yml:1]: Don't make a separate block, Put it under "CONTENT AUTHORING" block

File: src/Controller/CollaborativeEditorsController.php [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+ */
+class CollaborativeEditorsController extends ControllerBase {
+
+  /**
+   * Display the markup.
+   *
+   * @return array
+   *   Return markup array.
+   */
+  public function plugins() {
```
Comment (ankitjain28may) [inline src/Controller/CollaborativeEditorsController.php:18]: Show the plugins under this page like the Etherpad plugin else user won't be able to access the configuration page of the plugin.

File: modules/ce_etherpad/ce_etherpad.info.yml [hunk 6]
```diff
@@ -0,0 +1,8 @@
+name: Collaborative Editor Etherpad
+description: Allows integration with Etherpad.
+package: Collaborative Editors
+dependencies:
+  - collaborative_editors
+
```
Comment (ankitjain28may) [inline modules/ce_etherpad/ce_etherpad.info.yml:6]: Add the setting button directly here.

File: collaborative_editors.info.yml [hunk 7]
```diff
@@ -0,0 +1,6 @@
+name: Collaborative Editors
+description: Allows integration of third-party Collaborative Editor API.
+package: Collaborative Editors
+
```
Comment (ankitjain28may) [inline collaborative_editors.info.yml:4]: Add the setting button directly here.

**Review by ankitjain28may — CHANGES_REQUESTED** [review]
Remove the .module file as it is a blank file.

**Comment by anantjain6** [comment]
Continued in #2

**Comment by anantjain6** [comment]
Continued in #3

**Pull request closed without merging**